### ISSUE TITLE
Table of contents (support the --toc option)

### ISSUE THREAD

**Issue by pietsch** [mfenner/jekyll-pandoc#7]
I am looking for a way to create a table of contents for long pages. Pandoc supports this for standalone documents only:

````sh
pandoc -s --toc doc.md  # works
pandoc --toc doc.md  # does not work
````
This is intentional: [A closed pandoc ticket](https://github.com/jgm/pandoc/issues/156) explains that one could use standalone mode with a minimal custom pandoc template to get what I want.

I realize you are busy so I will try to come up with a patch.

**Comment by mfenner** [comment]
I am happy to look at a pull request. Given that I am in the process of converting the DataCite blog to a jekyll static site (see http://blog.datacite.org/adding-references-to-the-datacite-blog/), this is also of interest to me.

**Comment by pietsch** [comment]
I think I am halfway there. Not being a Ruby person, I do not know where to put the pandoc template toc.html (it has to be file) which looks like this:
````html
$if(toc)$
<div id="TOC">
$toc$
</div>
$endif$
$body$
````

For testing, I put the above into `/tmp/toc.html`. Now I can get tables of content by putting this into `_config.yml`:
````yaml
pandoc:
  extensions:
    - template: /tmp/toc.html
    - toc
````

**Comment by mfenner** [comment]
You can put the template anywhere, just specify the path as pandoc option (command line or jekyll _config.yml). I would create a `toc` folder, similar to the `bibliography` and `styles` folders I created for references.

**Comment by mfenner** [comment]
Another option is to put the template into the Ruby gem, I have done something similar here: https://github.com/mfenner/pandoc-jats-ruby, using a template for JATS. 

I actually have a similar use case: I want to generate both html and PDF versions with jekyll, and the default PDF template uses the YAML frontmatter without looking up author metadata in the `_data` folder. The easiest solution is again a custom template for PDF output.

**Comment by pietsch** [comment]
Yes, I have been trying to include the template into the Ruby gem, but I am still mystified about Ruby infrastructure tools.

**Comment by mfenner** [comment]
I can help you with that. The trick is to include non-Ruby files in the gemspec with something like https://github.com/mfenner/pandoc-jats-ruby/blob/master/pandoc-jats-ruby.gemspec#L25

**Comment by csillag** [comment]
Hey, are you still working in this? Turns out this exactly what I need for my stuff...

**Comment by mfenner** [comment]
@pietsch do you have this working? Happy to help finish this otherwise.

**Comment by pietsch** [comment]
Sorry, I have not found the time to learn the Ruby toolchain.

**Comment by mfenner** [comment]
If this is still something you want/need to do I can help with the Ruby part. I will need a toc for a project that starts in a few weeks.

**Comment by pietsch** [comment]
What other parts are there besides the Ruby part? Don't count on me hacking Haskell. I can test stuff, but I have no immediate need for this feature now.

**Comment by mfenner** [comment]
Ok, I will let you know in 2-3 months when I have worked on this.

**Comment by jpluimers** [comment]
Anyone still working on this? In `atom` having a reStructuredText preview without a `TOC` is a bit difficult to navigate.

Since there you cannot use the `--standalone` option, I'd be interested helping testing a solution that generates TOC without `-s`.

**Comment by mfenner** [comment]
@jpluimers the project I was thinking to use this for as taken a different turn. Would you be able to write a pull request?

**Comment by jpluimers** [comment]
@mfenner not really, as this is totally new territory for me.

In the mean time after restarting Atom, it now _does_ understand `--standalone`, so the `--toc` now works fine in https://atom.io/packages/rst-preview-pandoc
